src/libs/jira-lib.ts

```diff
@@ -0,0 +1,339 @@
+      const descendingPriorityIds = priorities.map(
+        (priority: { id: any }) => priority.id
+      );
+
+      return descendingPriorityIds;
+    } catch (error) {
+      throw new Error(`Error fetching priority IDs: ${error}`);
+    }
+  }
+  async createNewIssue(issue: Issue): Promise<Issue> {
+    try {
+      const assignee = process.env.ASSIGNEE ?? "";
+      if (assignee) {
+        issue.fields.assignee = { name: assignee };
+      }
+      issue.fields.project = { key: process.env.JIRA_PROJECT };
+
+      if (this.isDryRun) {
+        console.log(`[Dry Run] Would create a new issue with the following details:`, issue);
+
+        // Return a dry run issue object
+        this.dryRunIssueCounter++;
+        const dryRunIssue: Issue = {
+          id: `dryrun-id-${this.dryRunIssueCounter}`,
+          key: `DRYRUN-KEY-${this.dryRunIssueCounter}`,
+          fields: {
+            summary: issue.fields.summary || `Dry Run Summary ${this.dryRunIssueCounter}`,
+          },
+          webUrl: `${process.env.JIRA_BASE_URI}/browse/DRYRUN-KEY-${this.dryRunIssueCounter}`,
+        };
+
+        return dryRunIssue; // Return a dummy issue
+      }
+
+      const response = await this.axiosInstance.post('/rest/api/2/issue', issue);
+      const newIssue = response.data;
+      // Construct the webUrl for the new issue
+      newIssue["webUrl"] = `${process.env.JIRA_BASE_URI}/browse/${newIssue.key}`;
+      await this.removeCurrentUserAsWatcher(newIssue.key);
+      return newIssue;
+    } catch (error) {
+      throw new Error(`Error creating Jira issue: ${error}`);
+    }
+  }
+  async updateIssueTitleById(issueId: string, updatedIssue: Partial<Issue>) {
+    if (this.isDryRun) {
+      console.log(`[Dry Run] Would update issue title for issue ${issueId} with:`, updatedIssue);
+      return;
+    }
+
+    try {
+      const response = await this.axiosInstance.put(`/rest/api/2/issue/${issueId}`, updatedIssue);
+      console.log("Issue title updated successfully:", response.data);
+    } catch (error) {
+      throw new Error(`Error updating issue title: ${error}`);
+    }
+  }
+  async addCommentToIssueById(issueId: string, comment: string) {
+    if (this.isDryRun) {
+      console.log(`[Dry Run] Would add comment to issue ${issueId}:`, comment);
+      return;
+    }
+
+    try {
+      const response = await this.axiosInstance.post(`/rest/api/2/issue/${issueId}/comment`, { body: comment });
+      await this.removeCurrentUserAsWatcher(issueId); // Commenting on the issue adds the user as a watcher, so we remove them
+
+  
+    } catch (error) {
+      throw new Error(`Error adding comment to issue: ${error}`);
+    }
+  }
+  async findPathToClosure(transitions: any, currentStatus: string) {
+    const visited = new Set();
+    const queue: { path: string[]; status: string }[] = [
+      { path: [], status: currentStatus },
+    ];
+
+    while (queue.length > 0) {
+      const { path, status } = queue.shift()!;
```

Comment:
It looks like `queue` is always going to have a length of 1, so is there any reason why you can't just use `queue.pop()`, and remove the `while` loop? I think that may be a slightly more straightforward way of doing things?